### PR TITLE
Fix bug - Store::Subtract adds quantity instead of subtracting it

### LINKED ISSUE #751: `Store::Subtract` adds `quantity` instead of subtracting it
The fix for this should include at least one unit test, to make sure the code is fixed and working properly in this regard.

### PR DESCRIPTION
**Pull request by royfalk**
closes #751
Also add tests and a few additional methods to store.

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation **No. Only unit tests on store. Should be fine, as store code has not been linked to main game yet.**

### CODE REVIEW

File: engine/src/resource/store.cpp [hunk 1]
```diff
@@ -90,14 +90,32 @@ bool Store::Buy(Customer& seller, std::string product_name, double quantity)
...
         }
     }
 
     return false;
 }
 
+double Store::GetStock(std::string product_name) {
+    for(Product &in_store_product : stock) {
+        if(in_store_product == product_name ) {
+            return true;
```
Comment (BenjamenMeyer) [inline engine/src/resource/store.cpp:104]: shouldn't this be returning a number?

File: engine/src/resource/store.cpp [hunk 2]
```diff
@@ -90,14 +90,32 @@ bool Store::Buy(Customer& seller, std::string product_name, double quantity)
...
+        }
+    }
+
+    return -1;
```
Comment (BenjamenMeyer) [inline engine/src/resource/store.cpp:108]: Shouldn't this be `0` or is there some logic about it being `-1` for when there's no stock? If so, we should document that here too.

**Review by BenjamenMeyer — CHANGES_REQUESTED** [review]
Small bug - return double instead of double(boolean) for one code path.

**Review by stephengtuggy — APPROVED** [review]
Looks good to me. @BenjamenMeyer , can you re-review? TIA

File: engine/src/resource/store.cpp (hunk 2 above)
Comment (royfalk) [reply engine/src/resource/store.cpp:110]: It's an easy way to return an error, as product quantity is a resource and therefore, cannot be less than 0.

Comment (BenjamenMeyer) [reply engine/src/resource/store.cpp:110]: @royfalk thanks. We probably should make a standard for this across the code-base and ensure we get it in the documentation

**Review by BenjamenMeyer — APPROVED** [review]

**Pull request merged**